projects/RabbitMQ.Client/client/impl/WireFormatting.cs

```diff
@@ -34,13 +34,16 @@
 using System.Collections.Generic;
 using System.Runtime.CompilerServices;
 using System.Text;
+
 using RabbitMQ.Client.Exceptions;
 using RabbitMQ.Util;
 
 namespace RabbitMQ.Client.Impl
 {
     internal static class WireFormatting
     {
+        private static UTF8Encoding UTF8 = new UTF8Encoding();
```

Comment:
What made you add this change?  I've been wondering for some time now whether it's better to call Encoding.UTF8 vs a static instance. (Btw, should be readonly!)

From source I see they're not even the same instance 🤷‍♂️ 
=> https://source.dot.net/#System.Private.CoreLib/Encoding.cs,a10eb90a3d884500
=> https://source.dot.net/#System.Private.CoreLib/UTF8Encoding.Sealed.cs,98c4a50ea9fbaa13,references

Reply:
The Encoding.UTF8 returns the base class instead of an UTF8Encoding. Maybe there isn't any difference but this way we're at least pretty certain to avoid any callvirt calls. I saw this method used in other high-perf libraries as well.

Reply:
Quick test: (Code at the end)

GetBytes
|       Method |    value |     Mean |    Error |   StdDev | Ratio | Code Size |
|------------- |--------- |---------:|---------:|---------:|------:|----------:|
|     Original | MyString | 21.67 ns | 0.887 ns | 0.049 ns |  1.00 |     230 B |
|       Static | MyString | 21.40 ns | 0.710 ns | 0.039 ns |  0.99 |     230 B |
| StaticSealed | MyString | 20.85 ns | 0.800 ns | 0.044 ns |  0.96 |     230 B |

GetString
|       Method |   value |     Mean |    Error |   StdDev | Ratio | Code Size |
|------------- |-------- |---------:|---------:|---------:|------:|----------:|
|     Original | Byte[8] | 22.11 ns | 2.246 ns | 0.123 ns |  1.00 |     166 B |
|       Static | Byte[8] | 22.25 ns | 1.799 ns | 0.099 ns |  1.01 |     166 B |
| StaticSealed | Byte[8] | 22.11 ns | 1.126 ns | 0.062 ns |  1.00 |     166 B |

There's not much difference, only thing I see is a slight benefit for static sealed.

Benchmark code
```cs
using System.Collections.Generic;
using System.Text;
using BenchmarkDotNet.Attributes;

namespace Benchmarks.WireFormatting
{
    [ShortRunJob]
    [DisassemblyDiagnoser]
    public class EncodingBenchmarks
    {
        private static readonly UTF8Encoding UTF8 = new UTF8Encoding();
        private static readonly UTF8Encoding UTF8Sealed = (UTF8Encoding)Encoding.UTF8;

        public static IEnumerable<byte[]> ByteSource => new[] {Encoding.UTF8.GetBytes("MyString")};
        /*
        [Benchmark(Baseline = true)]
        [Arguments("MyString")]
        public byte[] Original(string value)
        {
            return Encoding.UTF8.GetBytes(value);
        }

        [Benchmark]
        [Arguments("MyString")]
        public byte[] Static(string value)
        {
            return UTF8.GetBytes(value);
        }

        [Benchmark]
        [Arguments("MyString")]
        public byte[] StaticSealed(string value)
        {
            return UTF8Sealed.GetBytes(value);
        }
        */
        [Benchmark(Baseline = true)]
        [ArgumentsSource(nameof(ByteSource))]
        public string Original(byte[] value)
        {
            return Encoding.UTF8.GetString(value);
        }

        [Benchmark]
        [ArgumentsSource(nameof(ByteSource))]
        public string Static(byte[] value)
        {
            return UTF8.GetString(value);
        }

        [Benchmark]
        [ArgumentsSource(nameof(ByteSource))]
        public string StaticSealed(byte[] value)
        {
            return UTF8Sealed.GetString(value);
        }
    }
}
```